drivers/ad717x/ad717x.c

```diff
@@ -439,6 +439,9 @@ int32_t AD717X_Init(ad717x_dev **device,
                 preg ++;
         }
 
+        /* Read ID register to identify the part */
+        ret = AD717X_ReadRegister(dev, AD717X_ID_REG);
```

Comment:
where is the ID register further used?

Reply:
this can also be problematic, since this changes the value of `ret` to non-zero, which can be interpreted as error

Reply:
ideally, you should also use the value read here;
either print it, or store it somewhere

Reply:
my bad here; i didn't realize that registers get stored in a regmap/reg list